MapboxCoreNavigation/NavigationServiceDelegate.swift

```diff
@@ -2,6 +2,20 @@ import Foundation
 import CoreLocation
 import MapboxDirections
 
+/**
+ NavigationServiceDelegate is the main delegation protocol in MapboxCoreNavigation. It's purpose is to delegate the operational parameters of the core navigation engine. See discussion for use-case examples.
+
+ You can use this protocol to:
+ * Recieve navigation progress updates
+ * Decide when to reroute the user
+ * Decide what location updates are considered 'qualified'
+ * Customize visual / spoken instructions that are presented to the user
+ * Decide if battery monitoring should be disabled
+ * Decide if the SDK should pause navigation when the user reaches an intermediate destination (for example, to show a interstitial modal UI)
+ * Be informed when the SDK fails to fetch a new route
+ * Be informed when the SDK is beginning or ending route-simulation
+ * Be informed when the user is approaching or has arrived at their destination
```

Comment:
This is a long list. A developer might as well look at the available methods to discover all this functionality. Meanwhile, it could be summarized here as:

> A navigation service delegate interacts with one or more `NavigationService` instances, such as `MapboxNavigationService` objects, during turn-by-turn navigation. This protocol is the main way that your application can synchronize its state with the SDK’s location-related functionality. Each of the protocol’s methods is optional.
> 
> As the user progresses along a route, a navigation service informs its delegate about significant events as they occur, and the delegate has opportunities to influence the route and its presentation. For example, when the navigation service reports that the user has arrived at the destination, your delegate implementation could present information about the destination. It could also customize individual visual or spoken instructions along the route by returning modified instruction objects.
> 
> Assign a `NavigationServiceDelegate` instance to the `NavigationService.delegate` property of the navigation service before you start the service.